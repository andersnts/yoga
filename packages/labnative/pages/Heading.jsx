import React from 'react';

import { Heading } from '@gympass/yoga';
import { Favorite, AddBooking } from '@gympass/yoga-icons';
import avatar from '../assets/images/avatarIcons/avatar.png';

const HeadingPage = () => (
  <Heading
    backButton={{
      onPress: () => {},
    }}
    rightIcons={[
      {
        icon: AddBooking,
        onPress: () => {},
      },
      {
        icon: Favorite,
        onPress: () => {},
      },
    ]}
    pageTitle="Page Title"
    title="Title"
    subtitle="Subtitle"
    avatar={avatar}
  />
);

export default HeadingPage;
