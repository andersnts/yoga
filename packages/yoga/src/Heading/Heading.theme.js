const Heading = ({ spacing }) => ({
  paddingTop: spacing.medium,
  paddingLeft: spacing.medium,
  paddingRight: spacing.medium,
  paddingBottom: spacing.large,
  rightIcons: {
    paddingLeft: spacing.large,
  },
  titleContainer: {
    marginTop: spacing.medium,
    avatar: {
      smallSize: spacing.xxlarge,
      largeSize: spacing.huge,
    },
  },
});

export default Heading;
