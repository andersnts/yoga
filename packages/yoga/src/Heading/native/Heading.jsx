import { arrayOf, bool, func, shape, string } from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { ArrowLeft } from '@gympass/yoga-icons';
import Box from '../../Box';
import Text from '../../Text';
import Button from '../../Button';
import Icon from '../../Icon';
import TitleContainer from './TitleContainer';

const HeadingContainer = styled(Box)`
  ${({
    theme: {
      yoga: {
        components: { heading },
      },
    },
  }) => `
    padding: ${heading.paddingTop}px ${heading.paddingRight}px ${heading.paddingBottom}px ${heading.paddingLeft}px;
`}
`;

const SubContainer = styled(Box)``;
const RightIconContainer = styled(Box)``;

const RightIcon = styled(Button.Link)`
  ${({
    isFirst,
    theme: {
      yoga: {
        components: { heading },
      },
    },
  }) => `
  padding-left: ${isFirst ? heading.rightIcons.paddingLeft : 0}px;
`}
`;
const PageTitle = styled(Text.Small)``;

const Heading = props => {
  const {
    backButton,
    pageTitle,
    rightIcons,
    title,
    subtitle,
    avatar,
    smallAvatar,
  } = props;

  return (
    <HeadingContainer w="100%">
      <SubContainer
        flexDirection="row"
        alignItems="center"
        justifyContent="space-between"
      >
        {backButton && (
          <Button.Link onPress={backButton.onPress}>
            <Icon
              as={ArrowLeft}
              width="medium"
              height="medium"
              fill="primary"
            />
          </Button.Link>
        )}
        <PageTitle>{pageTitle}</PageTitle>
        <RightIconContainer flexDirection="row">
          {rightIcons &&
            rightIcons.map((rightIcon, index) => (
              <RightIcon
                onPress={rightIcon.onPress}
                isFirst={index === 0}
                key={rightIcon.key}
              >
                <Icon
                  as={rightIcon.icon}
                  width="medium"
                  height="medium"
                  fill="primary"
                />
              </RightIcon>
            ))}
        </RightIconContainer>
      </SubContainer>
      {title && (
        <TitleContainer
          title={title}
          subtitle={subtitle}
          avatar={avatar}
          smallAvatar={smallAvatar}
        />
      )}
    </HeadingContainer>
  );
};

Heading.propTypes = {
  /** Add a back button to header */
  backButton: shape({
    onPress: func,
  }),
  /** Add a array with icons on right of header */
  rightIcons: arrayOf(
    shape({
      icon: func,
      onPress: func,
      key: string,
    }),
  ),
  /** Add a page title in header */
  pageTitle: string,
  /** Add a extra title in heading */
  title: string,
  /** Add a subtitle */
  subtitle: string,
  /** Add a avatar image */
  avatar: string,
  /** Change between a small or a large avatar */
  smallAvatar: bool,
};

Heading.defaultProps = {
  backButton: null,
  rightIcons: [],
  pageTitle: null,
  title: null,
  subtitle: null,
  avatar: null,
  smallAvatar: false,
};

export default Heading;
