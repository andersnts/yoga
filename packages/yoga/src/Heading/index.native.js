import Heading from './native';

export default Heading;
