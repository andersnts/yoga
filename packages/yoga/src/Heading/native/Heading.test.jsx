import React from 'react';
import { render } from '@testing-library/react-native';

import { AddBooking, Favorite } from '@gympass/yoga-icons';
import { ThemeProvider } from '../..';
import Heading from './Heading';

describe('<Heading />', () => {
  it('should create a complete heading', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading
          backButton={{
            onPress: jest.fn(),
          }}
          rightIcons={[
            {
              icon: AddBooking,
              key: 'AddBooking',
              onPress: jest.fn(),
            },
            {
              icon: Favorite,
              key: 'Favorite',
              onPress: jest.fn(),
            },
          ]}
          pageTitle="Page Title"
          title="Title"
          subtitle="Subtitle"
          avatar=""
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
  it('should create a heading without title', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading
          backButton={{
            onPress: jest.fn(),
          }}
          rightIcons={[
            {
              icon: AddBooking,
              key: 'AddBooking',
              onPress: jest.fn(),
            },
            {
              icon: Favorite,
              key: 'Favorite',
              onPress: jest.fn(),
            },
          ]}
          pageTitle="Page Title"
        />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
