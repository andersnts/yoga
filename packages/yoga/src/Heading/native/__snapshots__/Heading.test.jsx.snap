// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should create a complete heading 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 24,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
          "width": "100%",
        },
      ]
    }
    w="100%"
  >
    <View
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <TouchableWithoutFeedback>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            disabled={false}
            pressed={false}
            secondary={false}
            style={
              Array [
                Object {
                  "color": "#D8385E",
                  "fontFamily": "Rubik",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
              ]
            }
          >
            <RNSVGSvgView
              bbHeight={20}
              bbWidth={20}
              fill="#D8385E"
              focusable={false}
              height={20}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {
                      "height": 20,
                      "width": 20,
                    },
                  ],
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              width={20}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4292360286,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGSvgView>
          </Text>
        </View>
      </TouchableWithoutFeedback>
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            },
          ]
        }
      >
        Page Title
      </Text>
      <View
        flexDirection="row"
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TouchableWithoutFeedback
          isFirst={true}
          style={
            Array [
              Object {
                "paddingLeft": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              disabled={false}
              isFirst={true}
              pressed={false}
              secondary={false}
              style={
                Array [
                  Object {
                    "color": "#D8385E",
                    "fontFamily": "Rubik",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  Object {
                    "paddingLeft": 24,
                  },
                ]
              }
            >
              <RNSVGSvgView
                bbHeight={20}
                bbWidth={20}
                fill="#D8385E"
                focusable={false}
                height={20}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": 20,
                        "width": 20,
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                width={20}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4292360286,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          isFirst={false}
          style={
            Array [
              Object {
                "paddingLeft": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              disabled={false}
              isFirst={false}
              pressed={false}
              secondary={false}
              style={
                Array [
                  Object {
                    "color": "#D8385E",
                    "fontFamily": "Rubik",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  Object {
                    "paddingLeft": 0,
                  },
                ]
              }
            >
              <RNSVGSvgView
                bbHeight={20}
                bbWidth={20}
                fill="#D8385E"
                focusable={false}
                height={20}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": 20,
                        "width": 20,
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                width={20}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4292360286,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#231B22",
                "fontFamily": "Rubik",
                "fontSize": 24,
                "fontWeight": "500",
                "lineHeight": 32,
              },
            ]
          }
        >
          Title
        </Text>
        <Text
          color="deep"
          style={
            Array [
              Object {
                "color": "#6B6B78",
                "fontFamily": "Rubik",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 16,
              },
            ]
          }
        >
          Subtitle
        </Text>
      </View>
      
    </View>
  </View>
</View>
`;

exports[`<Heading /> should create a heading without title 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 24,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
          "width": "100%",
        },
      ]
    }
    w="100%"
  >
    <View
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <TouchableWithoutFeedback>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            disabled={false}
            pressed={false}
            secondary={false}
            style={
              Array [
                Object {
                  "color": "#D8385E",
                  "fontFamily": "Rubik",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
              ]
            }
          >
            <RNSVGSvgView
              bbHeight={20}
              bbWidth={20}
              fill="#D8385E"
              focusable={false}
              height={20}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {
                      "height": 20,
                      "width": 20,
                    },
                  ],
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              width={20}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4292360286,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGSvgView>
          </Text>
        </View>
      </TouchableWithoutFeedback>
      <Text
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 20,
            },
          ]
        }
      >
        Page Title
      </Text>
      <View
        flexDirection="row"
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TouchableWithoutFeedback
          isFirst={true}
          style={
            Array [
              Object {
                "paddingLeft": 24,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              disabled={false}
              isFirst={true}
              pressed={false}
              secondary={false}
              style={
                Array [
                  Object {
                    "color": "#D8385E",
                    "fontFamily": "Rubik",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  Object {
                    "paddingLeft": 24,
                  },
                ]
              }
            >
              <RNSVGSvgView
                bbHeight={20}
                bbWidth={20}
                fill="#D8385E"
                focusable={false}
                height={20}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": 20,
                        "width": 20,
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                width={20}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4292360286,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </Text>
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          isFirst={false}
          style={
            Array [
              Object {
                "paddingLeft": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              disabled={false}
              isFirst={false}
              pressed={false}
              secondary={false}
              style={
                Array [
                  Object {
                    "color": "#D8385E",
                    "fontFamily": "Rubik",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  Object {
                    "paddingLeft": 0,
                  },
                ]
              }
            >
              <RNSVGSvgView
                bbHeight={20}
                bbWidth={20}
                fill="#D8385E"
                focusable={false}
                height={20}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": 20,
                        "width": 20,
                      },
                    ],
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                width={20}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4292360286,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </View>
  </View>
</View>
`;
